### PR TITLE
Add NIC support

### PR DESCRIPTION
**Pull request by leopoldjuergen**
Details:
- Add NicManager class.
- Add Nic class.
- Add Unit Tests for NicManager and Nic.
- Add NicManager to Nic class.
- Add NIC to documentation reference.
- Add NIC example (example7.py).

### CODE REVIEW

File: zhmcclient/_nic.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A **NIC** object represents a single Network interface card (NIC)
+of a physical z Systems or LinuxONE computer that is in DPM mode
+(Dynamic Partition Manager mode).
```
Comment (andy-maier) [inline zhmcclient/_nic.py]: This sentencer suggests that the term NIC is used for the physical network adapter. The last sentence of this paragraph describes that better, and could be used as the definition of what a NIC is.

File: zhmcclient/_nic.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        Parameters:
+
+          properties (dict): Properties for the new NIC.
+            See the section in the :term:`HMC API` about the specific HMC
+            operation and about the 'Create NIC' description of the members
+            of the passed properties dict.
+
+        Returns:
+
+          string: The resource URI of the new NIC.
```
Comment (andy-maier) [inline zhmcclient/_nic.py]: Depending on the outcome of the discussion what to return here that we started for the Adapter resource, we should change the return value here as well.

File: zhmcclient/_nic.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    def create(self, properties):
+        """
+        Create and configures a NIC with the specified resource properties.
+
+        Parameters:
+
+          properties (dict): Properties for the new NIC.
+            See the section in the :term:`HMC API` about the specific HMC
+            operation and about the 'Create NIC' description of the members
+            of the passed properties dict.
```
Comment (andy-maier) [inline zhmcclient/_nic.py]: See my comment about phrasing such a reference, in the Adapters PR #45.

**Comment by andy-maier** [comment]
+1 - we agreed to defer the comments until later.

**Comment by leopoldjuergen** [comment]
+1

**Pull request merged**